src/main/java/ca/on/oicr/pineryreports/reports/impl/OctaneCountsReport.java

```diff
@@ -0,0 +1,508 @@
+  
+  private static final String DATE_REGEX = "\\d{4}-\\d{2}-\\d{2}";
+  private static final String DATE_FORMAT = "yyyy-MM-dd";
+  private static final String DATETIME_FORMAT = "yyyy-MM-dd HH:mm";
+  
+  private static final List<ColumnDefinition> COLUMNS = Collections.unmodifiableList(Arrays.asList(
+      new ColumnDefinition("CASE NUMBERS"),
+      new ColumnDefinition("")
+  ));
+  
+  private String start;
+  private String end;
+  
+  private final List<Count> caseNumbers = new ArrayList<>();
+  private final List<Count> sampleNumbers = new ArrayList<>();
+  private List<Count> inventoryNumbers = new ArrayList<>();
+  private List<Count> tissueTypeNumbers = new ArrayList<>();
+  private List<Count> tissueOriginNumbers = new ArrayList<>();
+  
+  @Override
+  public String getReportName() {
+    return REPORT_NAME;
+  }
+
+  @Override
+  public Collection<Option> getOptions() {
+    return Sets.newHashSet(OPT_AFTER, OPT_BEFORE);
+  }
+
+  @Override
+  public void processOptions(CommandLine cmd) throws ParseException {
+    if (cmd.hasOption(OPT_AFTER.getLongOpt())) {
+      String after = cmd.getOptionValue(OPT_AFTER.getLongOpt());
+      if (!after.matches(DATE_REGEX)) {
+        throw new ParseException("After date must be in format yyyy-mm-dd");
+      }
+      this.start = after;
+    }
+    
+    if (cmd.hasOption(OPT_BEFORE.getLongOpt())) {
+      String before = cmd.getOptionValue(OPT_BEFORE.getLongOpt());
+      if (!before.matches(DATE_REGEX)) {
+        throw new ParseException("Before date must be in format yyyy-mm-dd");
+      }
+      this.end = before;
+    }
+  }
+
+  @Override
+  public String getTitle() {
+    return "OCTANE Counts "
+        + (start == null ? "Any Time" : start)
+        + " - "
+        + (end == null ? "Now" : end)
+        + ", and Inventory "
+        + new SimpleDateFormat(DATE_FORMAT).format(new Date());
+  }
+  
+  private static final String ATTR_CATEGORY = "Sample Category";
+  private static final String ATTR_TISSUE_TYPE = "Tissue Type";
+  private static final String ATTR_TISSUE_ORIGIN = "Tissue Origin";
+  private static final String ATTR_SLIDES = "Slides";
+  private static final String ATTR_DISCARDS = "Discards";
+  private static final String ATTR_CONSUMED = "Slides Consumed";
+  private static final String ATTR_REMAINING = "Remaining";
+  private static final String ATTR_STAIN = "Stain";
+  
+  private static final String SAMPLE_CLASS_SLIDE = "Slide";
+  private static final String SAMPLE_CLASS_WHOLE_RNA = "whole RNA";
+  
+  private static final String SAMPLE_CATEGORY_IDENTITY = "Identity";
+  private static final String SAMPLE_CATEGORY_TISSUE = "Tissue";
+  private static final String SAMPLE_CATEGORY_STOCK = "Stock";
+  private static final String SAMPLE_CATEGORY_ALIQUOT = "Aliquot";
+
+  private static final String STAIN_HE = "Hematoxylin+Eosin";
+  
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 39 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)